packages/driver/src/cy/commands/navigation.ts

```diff
@@ -789,6 +789,15 @@ export default (Commands, Cypress, cy, state, config) => {
 
           const onBeforeLoad = (contentWindow) => {
             try {
+              // when using the visit the document referrer should be set to an empty string
+              if (config('modifyObstructiveCode')) {
```

Comment:
This doesn't need to be behind `modifyObstructiveCode`. `modifyObstructiveCode` controls if the `regex-rewriter` in the proxy layer runs or not, which is unrelated to this change to `document.referrer`. I think this fix is fine to apply to all `cy.visit()`s.

Reply:
this was suggested by @jennifer-shehane in the original issue description, will change 👍🏼

Reply:
@flotwig fixed in https://github.com/cypress-io/cypress/pull/18658/commits/568e9e58366aa934270b26c928c4794f7b2df700